src/NHibernate/SqlCommand/SqlString.cs

```diff
@@ -652,6 +652,18 @@ public SqlString SubstringStartingWithLast(string text)
 			return lastIndex >= 0 ? Substring(lastIndex) : Empty;
 		}
 
+		/// <summary>
+		/// Returns true if content is empty or white space characters only
+		/// </summary>
+		public bool IsEmptyOrWhitespace()
```

Comment:
Probably better to align with a `String` class: make it static; add support for `null`; call it `IsNullOrWhitespace`.

Reply:
I'm not sure it would give any benefits (at least from dev usability point of view). I tend to check instance methods for emptiness first (and sadly static methods are not even suggested). So I  would find this method quicker.

Reply:
But let majority rule :) @fredericDelaporte  Any preferences?

Reply:
Looking at it again, I see a third option: put that into `SqlStringHelper`. It has already an `IsEmpty` method (which is actually an `IsNullOrEmpty`), so it would be more consistent.

About discoverability, this can be improved by making it an extension method.

About aligning with `string` class, we would then have to move `IsEmpty` in `SqlString` too for consistency (and rename it by the way, of course).

Reply:
Some instance method is still required as I need to obtain data from `GetTrimmedIndexes`.
So I think I can add extension method  `IsNullOrWhitespace`  to 'SqlStringHelper' that will do null check + call `IsEmptyOrWhitespace` . Does it make any sense?

Reply:
From what I see instance method `IsEmptyOrWhitspace` needs to exists. So the main question does it improve anything adding additional wrappers over it.

Reply:
Leave it as is.

Reply:
I am not for having an `IsEmptyOrWhitespace` on instance along an `IsNullOrWhitespace` on helper. Better make `GetTrimmedIndexes` internal.

Reply:
> Leave it as is.

Simplest option. Agreed.